File: code/frontend/src/services/payment-service.ts
import api from "@/lib/axios";
import { PaymentRecordRequest, PaymentResponse, RefundRequest } from "@/types/backend";
import { PageResponse } from "./rental-service";

export const paymentService = {
  // POST /api/payments
  recordPayment: async (data: PaymentRecordRequest): Promise<PaymentResponse> => {
    const res = await api.post("/payments", data);
    return res.data;
  },

  // POST /api/payments/refunds
  recordRefund: async (data: RefundRequest): Promise<PaymentResponse> => {
    const res = await api.post("/payments/refunds", data);
    return res.data;
  },

  // GET /api/payments/rentals/{rentalId}
  getPaymentsByRental: async (
    rentalId: number,
    page = 0,
    size = 10
  ): Promise<PageResponse<PaymentResponse>> => {
    const res = await api.get(`/payments/rentals/${rentalId}`, { params: { page, size } });
    return res.data;
  },
};

File: code/backend/src/main/java/com/ioc/internship/service/PaymentService.java
package com.ioc.internship.service;

import com.ioc.internship.dto.request.PaymentRecordRequest;
import com.ioc.internship.dto.response.PaymentResponse;

public interface PaymentService {
    PaymentResponse recordPayment(PaymentRecordRequest request);
    PaymentResponse recordRefund(String email, com.ioc.internship.dto.request.RefundRequest request);
    
    org.springframework.data.domain.Page<PaymentResponse> getAllPayments(org.springframework.data.domain.Pageable pageable);
    org.springframework.data.domain.Page<PaymentResponse> getPaymentsByRental(Long rentalId, org.springframework.data.domain.Pageable pageable);
}
